mission_control/navigator_missions/navigator_singleton/navigator.py

```diff
@@ -74,6 +85,101 @@ def _load_vision_serivces(self, fname="vision_services.yaml"):
             s_client = self.nh.get_service_client(f[name]["topic"], s_type)
             self._vision_proxies[name] = {'client': s_client, 'request': s_req, 'args': f[name]['args']}
 
+
+class Searcher(object):
+    def __init__(self, nav, vision_proxy, search_pattern, **kwargs):
+        self.nav = nav
+        self.vision_proxy = vision_proxy
+        self.vision_kwargs = kwargs
+        self.search_pattern = search_pattern
+
+        self.object_found = False
+        self.response = None
+
+    def catch_error(self, failure):
+        if failure.check(defer.CancelledError):
+            print "SEARCHER - Cancelling defer."
+        else:
+            print "SEARCHER - There was an error."
+            print failure.printTraceback()
+            # Handle error
+
+    @util.cancellableInlineCallbacks
+    def start_search(self, timeout=60, loop=True, spotings_req=2, speed=.1):
+        print "SERACHER - Starting."
+        looker = self._run_look(spotings_req).addErrback(self.catch_error)
+        searcher = self._run_search_pattern(loop, speed).addErrback(self.catch_error)
```

Comment:
Looker and searcher sound really similar, to make this a little easier to read, I suggest maybe finder instead of looker and mover instead of searcher.